src/server/auth.ts

```diff
@@ -53,3 +55,43 @@ export default async function initializePatronTokenAuth(req: NextRequest) {
 
   return patronTokenResponse
 }
+
+/**
+ * Creates and returns redirect url from initializePatronTokenAuth() response
+ */
+export function getLoginRedirect(req) {
+  if (
+    !req.patronTokenResponse ||
+    !req.patronTokenResponse.isTokenValid ||
+    !req.patronTokenResponse.decodedPatron ||
+    !req.patronTokenResponse.decodedPatron.sub
+  ) {
+    const protocol = "http"
+    const hostname = req.headers["host"]
+    const originalUrl = BASE_URL + req.url
+    const fullUrl = encodeURIComponent(
+      `${protocol}://${hostname}${originalUrl}`
+    )
+    const redirect =
+      //`${appConfig.externalUrls.login}?redirect_uri=${fullUrl}`
+      `${process.env.LOGIN_BASE_URL}/login?redirect_uri=${fullUrl}`
+    console.log(redirect)
+    return redirect
+  }
+}
+
+/**
+ * Creates redirect to log out user, then return user to their current page. Requires an instance of useRouter.
+ */
+
+export function getLogoutRedirect(router) {
```

Comment:
The comment "Requires an instance of useRouter." is great. You can also make this required in the code by setting the type to `NextRouter` as `(router: NextRouter)`.

Reply:
(And import the type from "next/router")